packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -967,6 +967,24 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             metadata,
         );
 
+        const shouldRunExpiry = this.runtimeOptions.gcOptions?.gcAllowed === true
+        && this.runtimeOptions.gcOptions?.disableGC !== true
+        && this.runtimeOptions.gcOptions?.runSweep === true;
+
+        if(shouldRunExpiry) {
```

Comment:
You should add `shouldRunSweep` to `IGarbageCollector` (similar to shouldRunGC) and use that here.